src/lowtime_graph.rs

```diff
@@ -0,0 +1,214 @@
+use std::collections::{HashMap, HashSet};
+use ordered_float::OrderedFloat;
+use pathfinding::directed::edmonds_karp::{
+    SparseCapacity,
+    Edge,
+    EKFlows,
+    edmonds_karp,
+};
+
+
+#[derive(Clone)]
+pub struct LowtimeGraph {
+    node_ids: Vec<u32>,
+    source_node_id: Option<u32>,
+    sink_node_id: Option<u32>,
```

Comment:
I think I did this because of a weird implementation detail of `of_python` where the approach is to "start with an empty graph then `add_edge` for each edge".

In this approach, we can initialize the empty graph with a `source_node_id` and `sink_node_id` (even though we haven't added the nodes yet), and it would still run fine. But it felt philosophically weird to have an empty graph contains source/sink nodes that wasn't technically added to the graph yet.

Looking back, I think this small technicality probably isn't worth using an `Option`, so I'll just make them `u32`.